libraries/core-react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,193 @@
+import { forwardRef, SelectHTMLAttributes, useEffect, useState } from 'react'
+import {
+  useCombobox,
+  UseComboboxProps,
+  UseComboboxStateChange,
+} from 'downshift'
+import styled, { ThemeProvider } from 'styled-components'
+import { Label, Icon } from '../../'
+import { time } from '@equinor/eds-icons'
+import { spacingsTemplate } from '../../utils'
+import { timePicker as tokens } from './TimePicker.tokens'
+import {
+  Container,
+  PaddedInput,
+  StyledList,
+  StyledButton,
+  StyledListItem,
+  StyledInputWrapper,
+} from './commonStyles'
+import { useEds } from '../EdsProvider'
+import { useToken } from '../../hooks'
+
+const PaddedStyledListItem = styled(StyledListItem)`
+  ${({ theme }) => spacingsTemplate(theme.spacings)}
+`
+
+const defaultRange = [
+  '00:00',
+  '01:00',
+  '02:00',
+  '03:00',
+  '04:00',
+  '05:00',
+  '06:00',
+  '07:00',
+  '08:00',
+  '09:00',
+  '10:00',
+  '11:00',
+  '12:00',
+  '13:00',
+  '14:00',
+  '15:00',
+  '16:00',
+  '17:00',
+  '18:00',
+  '19:00',
+  '20:00',
+  '21:00',
+  '22:00',
+  '23:00',
+]
+
+export interface TimePickerProps {
+  /** Label for the select element */
+  label: string
+  className?: string
+  /** List of options to choose from */
+  timeRange?: string[]
+  value?: string
+  onValueChanged?: (time: string) => void
+  /** Disabled state */
+  disabled?: boolean
+  /** Read Only */
+  readOnly?: boolean
+}
+
+const TimePicker = forwardRef<HTMLDivElement, TimePickerProps>(
+  function SingleSelect(
```

Comment:
Copy pasta 😅 Make sure to change the name to `TimePicker`, or the debugger/tracelog might show wrong